tests/components/smlight/test_config_flow.py

```diff
@@ -0,0 +1,425 @@
+
+    mock_info = Info.from_dict(loads(load_fixture("smlight/info.json")))
+    with (
+        patch(
+            "pysmlight.web.Api2.get_info",
+            return_value=mock_info,
+        ),
+        patch(
+            "pysmlight.web.Api2.check_auth_needed",
+            return_value=True,
+        ),
+        patch(
+            "pysmlight.web.Api2.authenticate",
+        ),
+    ):
+        progress = hass.config_entries.flow.async_progress()
+        assert len(progress) == 1
+        assert progress[0].get("flow_id") == result["flow_id"]
+        assert progress[0]["context"].get("confirm_only") is True
+
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input={}
+        )
+
+        assert result2.get("type") is FlowResultType.FORM
+        assert result2.get("step_id") == "auth"
+
+        progress2 = hass.config_entries.flow.async_progress()
+        assert len(progress2) == 1
+        assert progress2[0].get("flow_id") == result["flow_id"]
+
+        result3 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={
+                CONF_USERNAME: "test-user",
+                CONF_PASSWORD: "test-pass",
+            },
+        )
+
+        assert result3.get("type") is FlowResultType.CREATE_ENTRY
+        assert result3 == snapshot
+
+        assert len(mock_setup_entry.mock_calls) == 1
+
+
+async def test_user_form(hass: HomeAssistant, mock_setup_entry: AsyncMock) -> None:
+    """Test we get the user form."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result.get("type") == FlowResultType.FORM
+    assert result.get("step_id") == "user"
+
+
+async def test_user_device_exists_abort(
+    hass: HomeAssistant, mock_config_entry: MockConfigEntry
+) -> None:
+    """Test we abort user flow if device already configured."""
+    mock_info = Info.from_dict(loads(load_fixture("smlight/info.json")))
+    with (
+        patch(
+            "pysmlight.web.Api2.get_info",
+            return_value=mock_info,
+        ),
+        patch(
+            "pysmlight.web.Api2.check_auth_needed",
+            return_value=False,
+        ),
+    ):
+        mock_config_entry.add_to_hass(hass)
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+            data={
+                CONF_HOST: "slzb-06.lan",
+            },
+        )
+
+        assert result.get("type") is FlowResultType.ABORT
+        assert result.get("reason") == "already_configured"
+
+
+async def test_zeroconf_device_exists_abort(
+    hass: HomeAssistant, mock_config_entry: MockConfigEntry
+) -> None:
+    """Test we abort zeroconf flow if device already configured."""
+    mock_info = Info.from_dict(loads(load_fixture("smlight/info.json")))
+    with (
```

Comment:
I believe we'll need a fixture here to avoid all those individual patches

Reply:
yes I agree, should have done that from the start, will implement